src/app/common/hooks/use-location-state.ts

```diff
@@ -5,15 +5,11 @@ import get from 'lodash.get';
 
 import { isUndefined } from '@shared/utils';
 
```

Comment:
These functions `useLocationState` + `useLocationStateWithCache` seemed to be declared multiple times, maybe by intention? 🤔 

I deleted the duplicates as it was affecting the type assertion

Reply:
These are [type overloads](https://www.typescriptlang.org/docs/handbook/2/functions.html#writing-good-overloads).

The logic goes, if it just accepts a prop, then it could be `string | undefined`, but if it has a default value passed, then it can safely be typed as `string`.